### DISCUSSION TITLE
issue: Validate defaultValues if not mounted on useForm()

### DISCUSSION THREAD

**Discussion by chamabreu** [react-hook-form/react-hook-form#7445] [Q&A] [answered]
### Version Number

7.22.5

### Codesandbox/Expo snack

https://codesandbox.io/s/unmountedfieldsvalidation-yssi5?file=/src/index.js

### Steps to reproduce

1. Go to [this Sandbox](https://codesandbox.io/s/unmountedfieldsvalidation-yssi5?file=/src/index.js)
2. Look the default Values in useForm, lastname is only 3 characters, but is validated with minLength 10
3. See the "Form is Valid" label, even if the defaultValue is "not valid"
4. Click the "Show LastName" Toggle in the Browser
5. the Lastname Field appers, and the "Form is Valid" switches to NOT Valid
6. But at this Point the console log does not contain the errors, where the invalid Field exists.
7. It needs a change in the lastName field to register the error.

### Expected behaviour

Even if the Field "LastName" is not registered at opening the Form, I need to validate the defaultValues (because they come from Backend).

This Codesandbox is simplified.
In my Project I have 3 "Tabs" which contain input fields, and only the Inpu fields on the active Tab gets validated.
I need validation for the whole form, even if the other fields are not mounted.

### What browsers are you seeing the problem on?

Chrome

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by bluebill1049** [comment]
This is by design and a feature that we only validate inputs that are mounted to the view. Separate your multiple step form to separate forms, so you can validate each step.

**Reply by yotamelkaslasy** [reply]
Hi @bluebill1049 I have a use case for this feature.
Currenlty i have multi route form that needs to register fields even when unmounted for validation or validate entire form based on default data.
It's not a stepper - it is a huge form splitted to sections with each section has a route.
I read everything and can't find a solution to this.
I found that i need to mount everything in advance, but then i get degregading performance.
I have 1 form wrapping the whole thing and everything works but validation of unmounted routes/sections.
Do you have a tip for me?
Thank you, appreciate the hard work you do with react hook form.

**Comment by Moshyfawn** [comment, marked as answer]
Hi, @yotamelkaslasy 
To validate all fields regardless whether or not they've been mounted, consider using form level validation with [@hookform/resolvers](https://github.com/react-hook-form/resolvers) by creating a validation schema and passing it into [a resolver](https://react-hook-form.com/api/useform#validationResolver).

To validate `defaultValues` you can use [a `trigger` form method](https://react-hook-form.com/api/useform/trigger) on form component mounting.

See [this CSB](https://codesandbox.io/s/validation-scema-example-l84s4) for example

**Reply by yotamelkaslasy** [reply]
Worked beatifully. Thank you so much @Moshyfawn
